opcua/common/node.py

```diff
@@ -404,15 +404,9 @@ def read_event_history(self, starttime=None, endtime=None, numvalues=0, evtypes=
         if not type(evtypes) in (list, tuple):
             evtypes = [evtypes]
 
-        # FIXME not a very nice way to make sure events.get_filter gets a list of nodes...
-        evtype_nodes = []
-        for evtype in evtypes:
-            if not isinstance(evtype, Node):
-                evtype_nodes.append(Node(self.server, ua.NodeId(evtype)))  # make sure we have a list of Node objects
-            else:
-                evtype_nodes.append(evtype)
+        evtypes = [Node(self.server, evtype) for evtype in evtypes]
 
```

Comment:
>I see you still have a comment about things to check in code. Will have a look.

What you did in node.py needs to be done in subscription.py. I think that's it. If I get time I'll add a test for custom event properties, but I'll probably do it in a separate PR. Don't have much free time at the moment.